GitHub.Authentication/Authentication.cs

```diff
@@ -336,5 +341,15 @@ public override void SetCredentials(TargetUri targetUri, Credential credentials)
         /// </param>
         /// <param name="result">The result of the interactive authentication attempt.</param>
         public delegate void AuthenticationResultDelegate(TargetUri targetUri, GitHubAuthenticationResultType result);
+
+        static Uri NormalizeUri(Uri targetUri)
+        {
+            // Special case for gist.github.com which are git backed repositories under the hood.
+            // Credentials for these repos are the same as the one stored with "github.com"
+            if (targetUri.DnsSafeHost.StartsWith("gist.", StringComparison.OrdinalIgnoreCase) && targetUri.DnsSafeHost.EndsWith(".github.com", StringComparison.OrdinalIgnoreCase))
```

Comment:
For `targetUri.DnsSafeHost.EndsWith(".github.com", StringComparison.OrdinalIgnoreCase)` would it make sense to swap out `".github.com"` with `GitHubBaseUrlHost`?

Reply:
Yeah, but I'd probably want to check `"." + GitHubBaseUrlHost` right? In case we get someone passing in `gist.badgithub.com`.

But now that I think about it, since this is a single special case, why don't I just check that `DnsSafeHost.Equal("gist.github.com", StringComparison.OrdinalIgnoreCase)`?